iceoryx_posh/test/moduletests/test_mepoo_memory_manager.cpp

```diff
@@ -341,13 +322,224 @@ TEST_F(MemoryManager_test, freeChunkMultiMemPoolFullToEmptyToFull)
+    constexpr uint32_t PADDING_BYTES{PAYLOAD_ALIGNMENT};
+
+    constexpr uint32_t EXPECTED_SIZE{sizeof(iox::mepoo::ChunkHeader) + CUSTOM_HEADER_SIZE + PADDING_BYTES
+                                     + PAYLOAD_SIZE};
+
+    auto chunkSize = iox::mepoo::MemoryManager::requiredChunkSize(
+        PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+
+    EXPECT_THAT(PADDING_BYTES, Ge(sizeof(iox::mepoo::ChunkHeader::PayloadOffset_t)));
+    EXPECT_THAT(chunkSize, Eq(EXPECTED_SIZE));
+}
+
+TEST_F(MemoryManager_test, requiredChunkSizeWithoutCustomPayloadAlignmentAndTooLargePayloadFails)
+{
+    constexpr uint32_t PAYLOAD_SIZE{std::numeric_limits<uint32_t>::max()};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{iox::CHUNK_DEFAULT_PAYLOAD_ALIGNMENT};
+    constexpr uint32_t CUSTOM_HEADER_SIZE{iox::CHUNK_NO_CUSTOM_HEADER_SIZE};
+    constexpr uint32_t CUSTOM_HEADER_ALIGNMENT{iox::CHUNK_NO_CUSTOM_HEADER_ALIGNMENT};
+
+    EXPECT_DEATH(
+        {
+            iox::mepoo::MemoryManager::requiredChunkSize(
+                PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+        },
+        ".*");
+}
+
+TEST_F(MemoryManager_test, requiredChunkSizeWithCustomPayloadAlignmentAndTooLargePayloadFails)
+{
+    constexpr uint32_t PAYLOAD_SIZE{std::numeric_limits<uint32_t>::max()};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{alignof(iox::mepoo::ChunkHeader) * 2};
+    constexpr uint32_t CUSTOM_HEADER_SIZE{iox::CHUNK_NO_CUSTOM_HEADER_SIZE};
+    constexpr uint32_t CUSTOM_HEADER_ALIGNMENT{iox::CHUNK_NO_CUSTOM_HEADER_ALIGNMENT};
+
+    EXPECT_DEATH(
+        {
+            iox::mepoo::MemoryManager::requiredChunkSize(
+                PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+        },
+        ".*");
+}
+
+TEST_F(MemoryManager_test, requiredChunkSizeWithCustomHeaderAndTooLargePayloadFails)
+{
+    constexpr uint32_t PAYLOAD_SIZE{std::numeric_limits<uint32_t>::max()};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{alignof(iox::mepoo::ChunkHeader) * 2};
+    constexpr uint32_t CUSTOM_HEADER_SIZE{8U};
+    constexpr uint32_t CUSTOM_HEADER_ALIGNMENT{8U};
+
+    EXPECT_DEATH(
+        {
+            iox::mepoo::MemoryManager::requiredChunkSize(
+                PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+        },
+        ".*");
+}
+
+TEST_F(MemoryManager_test, requiredChunkSizeWithCustomHeaderAlignmentLargerThanChunkHeaderAlignmentFails)
+{
+    constexpr uint32_t PAYLOAD_SIZE{0U};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{iox::CHUNK_DEFAULT_PAYLOAD_ALIGNMENT};
+    constexpr uint32_t CUSTOM_HEADER_SIZE{8U};
+    constexpr uint32_t CUSTOM_HEADER_ALIGNMENT{2 * alignof(iox::mepoo::ChunkHeader)};
+
+    EXPECT_DEATH(
+        {
+            iox::mepoo::MemoryManager::requiredChunkSize(
+                PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+        },
+        ".*");
+}
```

Comment:
I think the following tests are missing:
- PayloadSize = 42 with an payload alignment of zero (same for the custom header)
- PayloadSize = 42 with an uint32_t max payload alignment (same for the custom header)
- PayloadSize = 0 with an uint32_t max payload alignment (same for custom header)
- Custom Header Size = 0 with a reasonable payload alignment

Reply:
Should be done now. Note, a custom header alignment larger than the alignment of `ChunkHeader` is not allowed